### PR TITLE
add RadiumBlock endpoint service for Kusama

### PR DESCRIPTION
**Pull request by radiumb**
[RadiumBlock](https://radiumblock.com/) would like to contribute to the polkadot/kusama ecosystem by making our endpoint service for Kusama relay chain available to public!

### A little bit about [RadiumBlock](https://radiumblock.com/) :

RadiumBlock is a Web3 infrastructure service provider. We provide validator services for networks such as [Kusama](https://polkadot.js.org/apps/#/staking/targets?rpc=wss://kusama-rpc.polkadot.io&filter=RadiumBlock), [Polkadot](https://polkadot.js.org/apps/#/staking/targets?rpc=wss%3A%2F%2Frpc.polkadot.io&filter=RadiumBlock), [Polkadex](https://polkadot.js.org/apps/#/staking/targets?rpc=wss%3A%2F%2Fmainnet.polkadex.trade&filter=RadiumBlock), [Sora](https://polkadot.js.org/apps/#/staking/targets?rpc=wss%3A%2F%2Fmof2.sora.org&filter=RadiumBlock) etc.

Our team has over two decade of experience in managing compute infrastructure for large enterprises. We now harness our infrastructure expertise and experience to build reliable infrastructure services for the decentralized web3 future.
We participate in the Thousand Validator Program for Polkadot/Kusama and offer validator service to several other substrate chains. 

### A few words on our infrastructure: 

Our infrastructure is designed to be enterprise grade - with the ability to autoscale quickly to handle traffic bursts and provide services in different regions around the world to provide low latency services. Additionally, the infrastructure is designed to be resilient and fault tolerant even if some of the data centers in our service regions fail. The failover mechanism is designed patch the traffic over to the healthy data center regions.  

We believe its only befitting to the goal of Kusama as a canary network to launch our public endpoint first on Kusama. We have done several rounds of stress testing of our infrastructure and hope to learn more for this deployment.  

To ensure a stable infrastructure we have a 24/7 support staff monitoring the infrastructure with the help of several monitoring tools. Some of the key areas of observability we focus on are:

1. **_Endpoint availability and latency at different parts of the world._** 
We measure availability and latency from 28 points of observations around the world every 5 minutes. The system performs service monitoring by invoking a remote rpc call and an alarm is triggered if it takes more than 4 seconds to get a valid result.
Multi_region_endpoint_monitoring
2. **_Endpoint performance and load in each region of the world we deploy._** 
We have dashboards that observe and report various system performance, load and scaling parameters.  

Dashboard

Dashboard2

3. **_User experience of our Endpoint._** 
We use a browser based launch test that periodically loads our endpoint on one of the most demand pages of polkadot.js application (the staking page) - and observe from 6 different points in the world.
Multi_region_user_experience_monitoring

4. **_Monitor the blockchain stats of the nodes._** Finally we use our own dedicated telemetry server to monitor the blockchain stats of all our endpoint machines across different geographies.  
Telemetry

These monitoring tools are configured to trigger alerts to our support team.

### CODE REVIEW

**Comment by jacogr** [comment]
While I welcome new endpoints, I have to place a caution here - when my monitoring detects it has fallen over, it will be permanently removed ASAP. The reason is simple: Due to the random selection of endpoints, if enabled it will send 25% of all traffic in that direction. There has been at least 2 providers in the past that have been removed and they are not making it back in since they just couldn't cope and keep it up.

The overhead that I have to deal with in terms of support in terms of "my apps is not loading" is massive and it streams in across channels if an RPC provider misbehaves. So generally I am quite open to let new providers for Polkadot/Kusama into the UI, but at the first sign of trouble I do remove them immediately. (There is obviously leniency for proven-over-time providers, things _do_ and _will_ misbehave at certain points)

Having said that, I have a bit of trouble here -

(a) I have not seen the team active at all in terms of the Kusama/Polkadot ecosystem contributing
(b) There is absolutely no contract information or any information about the team

While I understand the need to not be "out in the open" on everything, this does raise red flags especially as a first stab at the highly used networks.

**Comment by radiumb** [comment]
[RadiumBlock](https://radiumblock.com) is certainly relatively new to the web3 space but we are veterans in compute infrastructure deployment and management aka web2. We are happy to have been exposed to web3 via the polkadot ecosystem and see a viable and sustainable path to contribute to infrastructure. It is our belief that decentralized services that allow the users to be in control is the future of the web and blockchain is the gateway to such a future.

### **Our journey**

Our journey into web3 and polkadot started with an investment in polkadex project as seed investors. Through that we have learned a great deal about the substrate based ecosystem. We are not blockchain developers rather our strength is in infrastructure management. In our traditional(web2) business Real Web LLC provides infrastructure and development services for some large client such as Variety magazine. Our infrastructure experience involves implementing high availability solutions and managing large cloud deployments on AWS, GPC and Cloudstack etc as well as owning and operating on prem server racks and data centers for ourselves as well as clients. 

You could say in blockchain we are a holder, validator, and infrastructure provider. 
 
### **Our contributions so far**

The endpoint is not our first contribution to kusama/polkadot ecosystem. As we got involved in the 1KV program we became aware of various infrastructure needs to the ecosystem. We responded to Will's (“Master of Validators” at Parity Technologies, and a co-founder of [Binary Holdings](https://www.binary.holdings/)) call to action for more providers of validator snapshots and we are now one of the 3 or 4 polkadot/kusama [validator snapshot](https://radiumblock.com/snapshots.html) provider. Our snapshots are being actively downloaded and we are in the process of making a formal proposal to the polkadot treasury to fund the service. 

Over the last year we have learned a lot about polkadot ecosystem and started to get involved by running validators in several substrate chains polkadex, kusama, polkadot, sora and hope to expand our presence and involvement. We are committed to expand our involvement in the polkadot ecosystem as an infrastructure provider. We are also providing validator node setup and maintenance as a service to some institutions that don't have the knowledge or capability to do it themselves.

### **Your support overhead** 

We completely understand the load on you in terms of support for an RPC provider, particularly should it misbehave. We anticipated this based on your past PR comments and that is why we proactively setup various monitoring system. Our company already has a team of 20 people supporting the web2 side of our business making it easier for us to ensure 24/7 monitoring of our web3 infrastructure. We have been stress testing various fail over and scaling scenarios for a month or so using load generators and rough estimates of load derived from past treasury proposals. We feel confident about RPC our implementation. **However** we are open to suggestions on beta testing our setup. May be a dry run from your end ? Or may be you can add our infrastructure to your monitoring for a fixed period before enabling it in the UI?

Given the burden you have with this role, it is absolutely fair for you to know about our team and more importantly a direct means to communicate with us. We would like to provide the team information and direct means to communicate including a direct contact number but we would prefer to do that directly rather quote that in a public forum. I have sent you our team and contact information via matrix.

Just like Dwellir and onFinality has been able to fund their public endpoint using the chain treasury, we feel confident our efforts too will be supported. If so, we foresee no issues in continuing the service. We are very excited to be involved in the infrastructure side of polkadot ecosystem and we hope you will give us an opportunity to be yet another provider - making the polkadot infrastructure service more vibrant. We understand we have to prove ourselves and its our hope we will have a chance to prove ourselves.

File: packages/apps-config/src/endpoints/productionRelayKusama.ts
Comment (jacogr) [inline packages/apps-config/src/endpoints/productionRelayKusama.ts:27]:
```suggestion
      RadiumBlock: 'wss://kusama.public.curie.radiumblock.co/ws',
```

Remove trailing spaces (should make the CI lint happy)

**Review by jacogr — CHANGES_REQUESTED** [review]

**Comment by jacogr** [comment]
I appreciate the comments above - the intent really is not to sound grumpy, but I guess that text comments do sometimes come over that way.

**Comment by radiumb** [comment]
Could you give us a sense of how much time after the PR is approved does the change go live and traffic start flowing? We just want to be extra prepared and vigilant when we start seeing the traffic flowing in.

**Comment by radiumb** [comment]
@jacogr I guess the workflows are waiting on your approval.

**Comment by radiumb** [comment]
@jacogr it says we requested a review but we didn't... assume it's part of the automated flow.

**Comment by radiumb** [comment]
@jacogr the merge is now upto you right? Please advise if there is anything needed from our end. Thanks much.

**Comment by jacogr** [comment]
I didn't want to merge Friday night, Saturday or Sunday just because of the traffic - don't expect everybody to be on-call for a go-live :) All good.

**Review by jacogr — APPROVED** [review]

**Comment by jacogr** [comment]
image

**Comment by radiumb** [comment]
Thank you @jacogr  !

**Comment by radiumb** [comment]
@jacogr any idea how long it normally take for large scale traffic to show up? We are seeing very light traffic but not yet seeing large scale traffic.

**Comment by jacogr** [comment]
The "select random endpoint" only kicks in when changing chains on the apps UI. Then once selected (and the user doesn't change it), it is kept for the next round, i.e. close the browser, open it, it is still used. A lot of people manually override as well to "their favorite" when changing.

**Pull request merged**
